src/tr/mangatr/src/eu/kanade/tachiyomi/extension/tr/mangatr/MangaTR.kt

```diff
@@ -116,4 +203,92 @@ class MangaTR : FMReader("Manga-TR", "https://manga-tr.com", "tr") {
 
     override fun pageListRequest(chapter: SChapter) =
         GET("$baseUrl/${chapter.url.substringAfter("cek/")}", headers)
+
+    override val pageListImageSelector = "div.chapter-content img.chapter-img"
+
+    // Manga-TR: image URL resolution with Base64-decoded data-src
+    override fun getImgAttr(element: Element?): String? {
+        return when {
+            element == null -> null
+            element.hasAttr("data-src") -> {
+                try {
+                    val encodedUrl = element.attr("data-src")
+                    val decodedBytes = Base64.decode(encodedUrl, Base64.DEFAULT)
+                    String(decodedBytes, StandardCharsets.UTF_8)
+                } catch (e: Exception) {
+                    element.attr("abs:src")
+                }
+            }
+            element.hasAttr("src") -> element.attr("abs:src")
+            element.hasAttr("data-original") -> element.attr("abs:data-original")
+            else -> null
+        }
+    }
+
+    // Simple pageListParse - relies on the selector above
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select(pageListImageSelector).mapIndexed { i, img ->
+            Page(i, document.location(), getImgAttr(img))
+        }
+    }
+
+    // =========================== List Parse ===========================
+    override fun popularMangaParse(response: Response): MangasPage {
+        ensureGenresLoaded()
+        return super.popularMangaParse(response)
+    }
+
+    private fun ensureGenresLoaded() {
+        if (cachedGenres.isNotEmpty()) return
+        runCatching {
+            val doc = client.newCall(GET("$baseUrl/manga-list.html", headers)).execute().asJsoup()
```

Comment:
This is called from `getFilterList()` so `execute()` might block the UI rendering thread. You should use `thread {}` to offload it to a background thread.